daprdocs/content/en/operations/extending/developing-grpc-components.md

```diff
@@ -0,0 +1,339 @@
+
+For this tutorial, you must have minimal knowledge of [gRPC and protocol buffers](https://grpc.io/), and that you chose a programming language [that supports gRPC](https://grpc.io/docs/languages/).
+
+For simplicity, all code samples uses the generated code from the [protoc](https://developers.google.com/protocol-buffers/docs/downloads) tool or supported gRPC build tools by languages.
+
+As previously mentioned, dapr uses a Unix Domain Socket to communicate with gRPC-based components, which means that as a prerequisite you also need a UNIX-like system, or for Windows users, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) should be sufficient.
+
+Also, install [.NET Core 6+](https://dotnet.microsoft.com/en-us/download)
+
+This tutorial is based on the [official Microsoft documentation](https://learn.microsoft.com/en-us/aspnet/core/grpc/basics?view=aspnetcore-6.0#c-tooling-support-for-proto-files).
+
+### Step 2: Downloading the base template project
+
+We've prepared a blueprint project that helps you to skip a few manual steps like cloning dapr repository and adding the required libraries. Get started by downloading and unzipping it using the following <a href="/code/dotnet-memstore.zip">link</a>.
+
+This template contains all the required changes to start developing a gRPC-based component from scratch, including protobuf definitions and an unimplemented InMemory StateStore that you are going to implement.
+
+### Step 3: Building and running
+
+At this point, you are ready to build and run the component. Let's test it out by running `dotnet build`
+
+```
+➜  DaprMemStoreComponent dotnet build
+MSBuild version 17.3.0+92e077650 for .NET
+  Determining projects to restore...
+  All projects are up-to-date for restore.
+  DaprMemStoreComponent -> [redacted]
+
+Build succeeded.
+    0 Warning(s)
+    0 Error(s)
+
+Time Elapsed 00:00:01.03
+```
+
+Great!
+
+Now, let's run the StateStore service by issuing a `dotnet run`
+
+```
+➜  DaprMemStoreComponent dotnet run
+Building...
+warn: Microsoft.AspNetCore.Server.Kestrel[0]
+      Overriding address(es) 'http://localhost:5259, https://localhost:7089'. Binding to endpoints defined via IConfiguration and/or UseKestrel() instead.
+info: Microsoft.Hosting.Lifetime[14]
+      Now listening on: http://unix:/tmp/dapr-components-sockets/memstore.sock
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Development
+info: Microsoft.Hosting.Lifetime[0]
+```
+
+To see it working, you can use the [grpc_cli](https://github.com/grpc/grpc/blob/master/doc/command_line_tool.md) tool for making gRPC calls. You can download and use it or use your preferred tool.
+
+Once properly downloaded and installed, open a new terminal instance, and let's invoke the `Features` method from the StateStore service, using the following command: `grpc_cli call unix:///tmp/dapr-components-sockets/memstore.sock Features ''`
```

Comment:
Out of curiosity: who is setting the DAPR_... env var? I am curious how the component got the `/tmp/dapr-coomponents-socket/memstore.sock` path. It wasn't given on the previous run step.

Reply:
from template